data-pipeline/src/trace_exporter.rs

```diff
@@ -98,25 +103,28 @@ fn add_path(url: &Uri, path: &str) -> Uri {
 /// Remove spans and chunks only keeping the ones that may be sampled by the agent
 fn drop_chunks(traces: &mut Vec<Vec<pb::Span>>) {
     traces.retain_mut(|chunk| {
+        // List of spans to keep even if the chunk is dropped
         let mut sampled_indexes = Vec::new();
         for (index, span) in chunk.iter().enumerate() {
+            // ErrorSampler
             if span.error == 1 {
                 // We send chunks containing an error
                 return true;
             }
-            let priority = span.metrics.get("_sampling_priority_v1");
-            if priority.is_some_and(|p| *p > 0.0) {
-                if has_top_level(span) {
-                    // We send chunks with positive priority
-                    return true;
-                }
-                // We send single spans with positive priority
-                sampled_indexes.push(index);
-            } else if priority.is_none() && has_top_level(span) {
-                // We send chunks with no priority
+            // PrioritySampler and NoPrioritySampler
+            let priority = span.metrics.get(SAMPLING_PRIORITY_KEY);
+            if has_top_level(span) && (priority.is_none() || priority.is_some_and(|p| *p > 0.0)) {
```

Comment:
minor: Do you think `priority.map_or(true, |p| *p > 0.0)` is clearer than `(priority.is_none() || priority.is_some_and(|p| *p > 0.0))`?

Reply:
Also, do we have sufficient unit tests covering this logic here? I didn't see any changes to tests.

Reply:
There are tests [here](https://github.com/DataDog/libdatadog/blob/321e98ecd1be000873ec1ba8831977cc6787683f/data-pipeline/src/trace_exporter.rs#L989). The sampling logic has not changed so no test update required.

The `(priority.is_none() || priority.is_some_and(|p| *p > 0.0))` syntax seems clearer to me